pkg/operators/ebpf/ebpf.go

```diff
@@ -639,13 +644,11 @@ func (i *ebpfInstance) Start(gadgetCtx operators.GadgetContext) error {
 				continue
 			}
 			i.logger.Debugf("setting var %q to %v", v.name, t)
-			constReplacements[v.name] = res
-		}
-	}
 
-	//nolint:staticcheck
-	if err := i.collectionSpec.RewriteConstants(constReplacements); err != nil {
-		return fmt.Errorf("rewriting constants: %w", err)
+			if err := ebpfutils.SpecSetVar(i.collectionSpec, v.name, res); err != nil {
```

Comment:
`ebpfutils.SpecSetVar` is called twice in this file: for gadget parameters and for ebpf instance variables. Could it be called twice for the same cilium/ebpf variable? If so, a comment explaining the precedence we've chosen would help.

Reply:
The second call is done to set some internal variables like enabling filtering by mount ns:

 https://github.com/inspektor-gadget/inspektor-gadget/blob/8912ea06de042187f4a125402694dd17b2ed7edd/pkg/operators/localmanager/localmanager.go#L610

Unless there is a mistake, SpecSetVar won't be called twice for the same variable, since params are defined by users with the GADGET_PARAM macro, it shouldn't overlap with "internal variables".